app/modules/seed_library/endpoints_seed_library.py

```diff
@@ -0,0 +1,506 @@
+    """
+    if plant_base.ancestor_id is None:
+        if not is_user_member_of_any_group(
+            user=user,
+            allowed_groups=[GroupType.seed_library],
+        ):
+            raise HTTPException(
+                status_code=403,
+                detail="Please contact an Planet&co member if you are trying to donate a plant that does not come from the seed library",
+            )
+
+    species_reference = await cruds_seed_library.get_species_by_id(
+        plant_base.species_id,
+        db,
+    )
+    if species_reference is None:
+        raise HTTPException(
+            404,
+            "Species not found",
+        )
+    date = datetime.now(tz=UTC)
+    if species_reference:
+        plant_reference = (
+            str(species_reference.prefix)
+            + str(date.day)
+            + str(date.month)
+            + str(date.year)
+        )
+        plant_number = await cruds_seed_library.count_plants_created_today(
+            plant_reference,
+            db,
+        )
+
+        plant_reference = (
+            plant_reference + "0" * (3 - len(str(plant_number))) + str(plant_number)
+        )
+
+    plant = schemas_seed_library.PlantComplete(
+        id=uuid.uuid4(),
+        state=plant_base.state,
+        species_id=plant_base.species_id,
+        propagation_method=plant_base.propagation_method,
+        nb_seeds_envelope=plant_base.nb_seeds_envelope,
+        plant_reference=plant_reference,
+        ancestor_id=plant_base.ancestor_id,
+        previous_note=plant_base.previous_note,
+        current_note=plant_base.current_note,
+        borrower_id=plant_base.borrower_id,
+        confidential=plant_base.confidential,
+        nickname=plant_base.nickname,
+        planting_date=plant_base.planting_date,
+        borrowing_date=plant_base.borrowing_date,
+    )
+
+    await cruds_seed_library.create_plant(plant, db)
+    return plant
+
+
+@module.router.patch(
+    "/seed_library/plant/{plant_id}",
+    status_code=204,
+)
+async def update_plant(
+    plant_id: uuid.UUID,
+    plant_edit: schemas_seed_library.PlantEdit,
+    user: models_core.CoreUser = Depends(is_user()),
+    db: AsyncSession = Depends(get_db),
+):
+    """
+    Update a Plant
+    **This endpoint is only usable by the owner of the plant**
+    """
+    plant = await cruds_seed_library.get_plant_by_id(plant_id, db)
+    if plant is None:
+        raise HTTPException(404, "Plant does not exist.")
+
+    if plant.borrower_id != user.id:
+        raise HTTPException(
+            status_code=403,
+            detail=f"You are not allowed to update plant {plant_id} since you do not own the plant",
+        )
+    try:
+        await cruds_seed_library.update_plant(
+            plant_id=plant_id,
+            plant_edit=plant_edit,
+            db=db,
+        )
+    except ValueError as error:
+        raise HTTPException(status_code=400, detail=str(error))
```

Comment:
I think `update_plant` should never raise a ValueError.

By catching the error raised by `update_plant` and returning an error 400, the error will never be logged.
With the following code, FastAPI will catch the exception itself, log it and then return an error 500:
```suggestion
    await cruds_seed_library.update_plant(
            plant_id=plant_id,
            plant_edit=plant_edit,
            db=db,
        )
```